### PR TITLE
addon: logging improvements

### PR DESCRIPTION
**Pull request by JoaoBraveCoding**
Tested ✅ 

Screenshot:
image

### CODE REVIEW

**Review by periklis — APPROVED** [review]

File: internal/addon/authentication/certificates.go [hunk 1]
```diff
@@ -24,24 +24,34 @@ func CreateOrUpdateRootCertificate(k8s client.Client) error {
...
 		switch op {
 		case ctrlutil.OperationResultNone:
-			klog.Info(msg)
+			l.Info(msg)
 		default:
-			klog.Info(msg)
+			l.Info(msg)
 		}
 	}
+	if errCount > 0 {
```
Comment (iblancasa) [inline internal/addon/authentication/certificates.go:52]: You can just enable a flag or initialize an `err` variable, right?

File: internal/addon/authentication/provider.go [hunk 2]
```diff
@@ -88,24 +90,34 @@ func (sp *secretsProvider) GenerateSecrets(targetAuthType map[string]string) (ma
...
 
-		msg := fmt.Sprintf("Resource has been %s", op)
+		msg := fmt.Sprintf("resource has been %s", op)
 		switch op {
 		case ctrlutil.OperationResultNone:
-			klog.Info(msg)
+			l.Info(msg)
 		default:
-			klog.Info(msg)
+			l.Info(msg)
```
Comment (iblancasa) [inline internal/addon/authentication/provider.go:115]: Why do you keep both?

File: internal/addon/authentication/provider.go [hunk 3]
```diff
@@ -88,24 +90,34 @@ func (sp *secretsProvider) GenerateSecrets(targetAuthType map[string]string) (ma
...
 		op, err := ctrl.CreateOrUpdate(ctx, sp.k8s, obj, mutateFn)
 		if err != nil {
-			klog.Error(err, "failed to configure resource")
+			l.Error(err, "failed to configure resource")
+			errCount++
 			continue
 		}
 
-		msg := fmt.Sprintf("Resource has been %s", op)
+		msg := fmt.Sprintf("resource has been %s", op)
```
Comment (iblancasa) [inline internal/addon/authentication/provider.go:110]: Adding the name (maybe the namespace) of the resource can be helpful for the future

Comment (JoaoBraveCoding) [reply internal/addon/authentication/provider.go:110]: You can see in the screenshot of the description of the PR what the logs look like. If you notice the name of the resource is already there. We could potentially add namespace but for the resources created in this function I don't think they'll be o much help

File: internal/addon/authentication/certificates.go (hunk 1 above)
Comment (JoaoBraveCoding) [reply internal/addon/authentication/certificates.go:52]: I've added the errorCount to the message

File: internal/addon/authentication/provider.go (hunk 3 above)
Comment (JoaoBraveCoding) [reply internal/addon/authentication/provider.go:111]: Added namespace

File: internal/addon/authentication/provider.go (hunk 2 above)
Comment (JoaoBraveCoding) [reply internal/addon/authentication/provider.go]: I was missing the logging level

**Comment by JoaoBraveCoding** [comment]
Closing this PR as future improvements in this area will be done in a future PR

**Pull request closed without merging**